### ISSUE TITLE
When adding an import based on a completion to a unit test it becomes an unused import in the normal build

### ISSUE THREAD

**Issue by Lokathor** [rust-lang/rust-analyzer#12661]
The problem is that if you have some code like
```rust
#[test]
fn test_all_room_ids_different() {
  let mut set = HashS
}
```
and then you press Ctrl+Space at the end of the line and pick `std::collections::HashSet` as the completion, it'll add the `std::collections::HashSet` to the top of the entire module. Except if nothing else in the module is using a HashSet then the non-test build now marks the import as an unused import.

I don't know what the good answer is here, but there are two options that come to mind:
1) put the new import at the top of the test function
2) put the new import at the top of the module behind `#[cfg(test)]`

**Comment by jhgg** [comment]
The general solution to this is to put your tests inside a module, e.g.:
```rust
#[cfg(test)]
mod tests {
    #[test]
    fn test_whatever() {
          let mut set = HashS$0
    }
}
```

This will put the import inside the test module.

**Comment by Veykril** [comment]
Ye we should special case `#[test]` items a bit, we already insert imports into the scope of a `cfg`ed item for now (as that was simpler to implement than importing with a `cfg` attacehd to the import), I think we can do the same here for the time being, treating `#[test]` as `#[cfg(test)]`.

**Comment by flodiebold** [comment]
I think if we started doing that for all `#[test]` items, even those in test modules, it would get annoying fast. Hopefully we can only do it if the surrounding module isn't `#[cfg(test)]`?

**Comment by Veykril** [comment]
Yes, sorry I meant the case where the items are not inside a cfg(test) module.
